drivers/counter/counter_mchp_xec.h

```diff
@@ -0,0 +1,78 @@
+/* counter_xec.h - Microchip XEC counter driver private definitions */
+
+/*
+ * Copyright (c) 2019 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_COUNTER_COUNTER_XEC_H_
+#define ZEPHYR_DRIVERS_COUNTER_COUNTER_XEC_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct counter_xec_config {
+	struct counter_config_info info;
+	void (*config_func)(void);
+	u32_t base_address;
+	u16_t prescaler;
+	u8_t girq_id;
+	u8_t girq_src;
+};
+
+struct counter_xec_data {
+	counter_alarm_callback_t alarm_cb;
+	counter_top_callback_t top_cb;
+	void *user_data;
+};
+
+/* See datasheet Section 19.9 */
+
+struct counter_xec_registers {
+	volatile u32_t count_reg;
+	union {
```

Comment:
@ceolin can you share where we have posted coding style guidelines about anonymous structs/unions?

Reply:
we still haven't post it. We had a discussion a long time ago in a TSC meeting about c11xc99, but stakeholders never made a decision. Now it seems that C99 is a easier path (in the sense of less uncertainties) for FuSA.

Considering that c99 is already being set on Zephyr, unnamed structs/unions are not allowed by this standard and should not be used.

Reply:
> we still haven't post it.

Then post it asap, please.
I am fine with following any code style, standard and so on as long as it is officially documented.

Will apply the change

Reply:
We are figuring out the proper way to distribute it since MISRA-C has its copyrights.
But anyway, this special case is violating c99 which is the standard used by Zephyr (though we have many other violations). We have to avoid gnu extensions or features from others standards like c11.

Reply:
I see, thanks for the background details